valhalla/thor/dijkstras.h

```diff
@@ -0,0 +1,273 @@
+#ifndef VALHALLA_THOR_Dijkstras_H_
+#define VALHALLA_THOR_Dijkstras_H_
+
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <valhalla/baldr/double_bucket_queue.h>
+#include <valhalla/baldr/graphid.h>
+#include <valhalla/baldr/graphreader.h>
+#include <valhalla/baldr/location.h>
+#include <valhalla/proto/tripcommon.pb.h>
+#include <valhalla/sif/dynamiccost.h>
+#include <valhalla/sif/edgelabel.h>
+#include <valhalla/thor/edgestatus.h>
+
+namespace valhalla {
+namespace thor {
+
+enum class InfoEdgeType {
+  origin,
+  destination,
+  regular,
+};
+enum class InfoRoutingType {
+  forward,
+  bidirectional,
+  multi_modal,
+};
+
+struct ExpandingNodeMiscInfo {
+  InfoEdgeType edge_type;
+  InfoRoutingType routing_type;
+  bool from_transition;
+};
+
+enum class RouteCallbackRecommendedAction {
+  no_action,
+  skip_expansion,
+  stop_expansion,
+};
+
+/**
+ * Algorithms to do shortest first graph traversal
+ */
+class Dijkstras {
+public:
+  /**
+   * Constructor.
+   */
+  Dijkstras();
+
+  /**
+   * Destructor
+   */
+  virtual ~Dijkstras();
+
+  /**
+   * Clear the temporary memory (adjacency list, edgestatus, edgelabels)
+   */
+  void Clear();
+
+  /**
+   * Compute the best first graph traversal from a list of origin locations
+   * @param  origin_locs  List of origin locations.
+   * @param  graphreader  Graphreader
+   * @param  mode_costing List of costing objects
+   * @param  mode         Travel mode
+   */
+  void Compute(google::protobuf::RepeatedPtrField<valhalla::Location>& origin_locs,
+               baldr::GraphReader& graphreader,
+               const std::shared_ptr<sif::DynamicCost>* mode_costing,
+               const sif::TravelMode mode);
+
+  /**
+   * Compute the best first graph traversal to a list of destination locations
+   * @param  origin_locs  List of origin locations.
```

Comment:
s/origin_locs/dest_locactions

Also, for consistency, lets use either the full or abbreviated form for `locations`.